From dad1390e253f010df8de8fa816b889523b249d01 Mon Sep 17 00:00:00 2001
From: Laszlo Agocs <lagocs83@gmail.com>
Date: Mon, 29 Sep 2025 09:40:09 +0200
Subject: [PATCH 114/239] QQuickRhiItem docs: follow QRhiWidget docs wrt fixed
 size

Pick-to: 6.9 6.8
Change-Id: Ia212eabc2868714fc2065badedf12332c8a92642
Reviewed-by: Andy Nichols <andy.nichols@qt.io>
(cherry picked from commit fea9a8bcb240efce713389171aa594d8916de784)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quick/items/qquickrhiitem.cpp | 30 ++++++++++++++++++++++--------
 1 file changed, 22 insertions(+), 8 deletions(-)

diff --git a/src/quick/items/qquickrhiitem.cpp b/src/quick/items/qquickrhiitem.cpp
index c99a8eee8e..bd69931977 100644
--- a/src/quick/items/qquickrhiitem.cpp
+++ b/src/quick/items/qquickrhiitem.cpp
@@ -722,13 +722,29 @@ void QQuickRhiItem::setMirrorVertically(bool enable)
     item's area.
 
     For example, setting a size that is exactly twice the item's (pixel) size
-    effectively performs 2x supersampling (rendering at twice the resolution
-    and then implicitly scaling down when texturing the quad corresponding to
-    the item in the scene).
+    effectively performs 2x supersampling (rendering at twice the resolution and
+    then implicitly scaling down when texturing the quad corresponding to the
+    item in the scene). On the other hand, setting a size that is half of the
+    item's pixel size effectively achieves rendering at half resolution and then
+    upscaling the results.
 
     By default the value is \c 0. A value of 0 means that texture's size
     follows the item's size. (\c{texture size} = \c{item size} * \c{device
     pixel ratio}).
+
+    \note The device pixel ratio (the system compositor's scale factor) can have
+    a big impact on performance, since a scale factor of 2 (200%) means
+    rendering at twice the resolution, so twice of what the developer and UI
+    designer perceives as the item's size, and then effectively downscaling the
+    content, similarly to what happens when setting this property to twice the
+    item's pixel size on a system where the device pixel ratio is 1. Therefore,
+    this property is expected to be rarely used with sizes bigger than the
+    item's pixel size, since many modern desktop systems have neither the need
+    nor have the performance budget for it, when a larger than 1 device pixel
+    ratio is used anyway by the system. Instead, the main use case for this
+    property is to set a smaller size, in order to render at a reasonable
+    smaller resolution instead of blindly following the item (and perhaps
+    window) geometry, however big that may be.
  */
 int QQuickRhiItem::fixedColorBufferWidth() const
 {
@@ -756,14 +772,12 @@ void QQuickRhiItem::setFixedColorBufferWidth(int width)
     within the scene), which means the texture's content will appear stretched
     (scaled up) or scaled down onto the item's area.
 
-    For example, setting a size that is exactly twice the item's (pixel) size
-    effectively performs 2x supersampling (rendering at twice the resolution
-    and then implicitly scaling down when texturing the quad corresponding to
-    the item in the scene).
-
     By default the value is \c 0. A value of 0 means that texture's size
     follows the item's size. (\c{texture size} = \c{item size} * \c{device
     pixel ratio}).
+
+    See \l fixedColorBufferWidth for more information on the use cases for
+    setting a fixed width and height.
  */
 
 int QQuickRhiItem::fixedColorBufferHeight() const
-- 
2.51.2

